### PR TITLE
fix data in RasaModelData

### PR DESCRIPTION
**Pull request by Ghostvv**
**Proposed changes**:
- ...

**Status (please check what you already did)**:
- [ ] added some tests for the functionality
- [ ] updated the documentation
- [ ] updated the changelog (please check [changelog](https://github.com/RasaHQ/rasa/tree/master/changelog) for instructions)
- [ ] reformat files using `black` (please check [Readme](https://github.com/RasaHQ/rasa#code-style) for instructions)

### CODE REVIEW

File: rasa/utils/tensorflow/tf_layers.py [hunk 1]
```diff
@@ -373,7 +371,7 @@ def call(
 
 
 class InputMask(tf.keras.layers.Layer):
-    def build(self, input_shape: List[int]) -> None:
+    def build(self, input_shape: "tf.TensorShape") -> None:
```
Comment (tabergma) [inline rasa/utils/tensorflow/tf_layers.py]:
```suggestion
    def build(self, input_shape: tf.TensorShape) -> None:
```

File: rasa/utils/tensorflow/tf_model_data.py [hunk 2]
```diff
@@ -36,6 +40,36 @@ def keys(self):
     def feature_not_exists(self, key: Text) -> bool:
         return key not in self.data or not self.data[key]
 
+    def get_number_of_examples(self) -> int:
+        """Obtain number of examples in session data.
```
Comment (tabergma) [inline rasa/utils/tensorflow/tf_model_data.py]:
```suggestion
        """Obtain number of examples in data.
```

File: rasa/utils/tensorflow/tf_model_data.py [hunk 3]
```diff
@@ -185,61 +220,23 @@ def balance(self, batch_size: int, shuffle: bool) -> None:
...
-        """Get the feature dimension of the given key."""
-
-        number_of_features = 0
-        for data in self.data[key]:
-            if data.size > 0:
-                number_of_features += data[0].shape[-1]
-
-        return number_of_features
-
-    def convert_to_tf_dataset(
```
Comment (tabergma) [inline rasa/utils/tensorflow/tf_model_data.py]: was this a duplicate?

File: rasa/utils/tensorflow/tf_models.py [hunk 4]
```diff
@@ -59,20 +58,24 @@ def train_dataset_function(_batch_size):
...
 
         if eager:
             # allows increasing batch size
             tf_train_dataset_function = train_dataset_function
             tf_train_on_batch_function = self.train_on_batch
         else:
+            logger.debug("Building tensorflow train graph...")
             # allows increasing batch size
             tf_train_dataset_function = tf.function(func=train_dataset_function)
+            init_dataset = tf_train_dataset_function(tf_batch_size)
```
Comment (tabergma) [inline rasa/utils/tensorflow/tf_models.py]: Why did you add this?

File: rasa/utils/tensorflow/tf_model_data.py (hunk 3 above)
Comment (Ghostvv) [reply rasa/utils/tensorflow/tf_model_data.py]: yes, it did the same thing as `as_tf_dataset` and was never called

File: rasa/utils/tensorflow/tf_models.py (hunk 4 above)
Comment (Ghostvv) [reply rasa/utils/tensorflow/tf_models.py]: to prebuild tf graph before tqdm starts, separate building and running time. it is a part of the next PR

Comment (tabergma) [reply rasa/utils/tensorflow/tf_models.py]: ok, great. thanks for clarifying.
